exercises/IaC/ansible/variables/.templates/README.md.erb

```diff
@@ -0,0 +1,378 @@
+
+### Variables files
+<%
+prod_yaml =<<YAML
+---
+prompt: PROD
+YAML
+
+dev_yaml =<<YAML
+---
+prompt: DEV
+YAML
+%>
+
+Not every system that you are installing will be identical. Take for example the scenario where you are setting up the development and production environments for a continuous deployment pipeline. It can be useful for these environments to be subtly different. For example maybe having the commandline prompt differentiate an environment as either belonging to development or production.
+
+Variables are a good way of managing the differences between target systems without the need to introduce logic, as we will now see.
+
+<%
+dev_server = 'dev1'
+%>
+
+When `<%= cic_up_command %>` was run, the following environments with given hostnames were  stood up:
+
+- <%= dev_server %>
+- prod1
+- prod2
+
+To connect to any of these environment simply use the `<%=cic_connect_command %>` connect command, e.g.  `<%= cic_connect_command %> <%= dev_server %>`
+
+
+Write the following development specific config to `<%= write_to_file('ansible/vars/dev.yml', dev_yaml) %>`
+```YAML
+<%= dev_yaml %>
+```
+
+
+Write the following production specific config to `<%= write_to_file('ansible/vars/prod.yml', prod_yaml) %>`
+```YAML
+<%= dev_yaml %>
+```
+
+Each of the files defines a variable `prompt` that will be displayed on the console.
+
+**Note** There is nothing special about the folder name `vars`, its purely arbitrary.
+
+<%
+dev_prompt= 'dev'
+prompt_playbook=<<YAML
+---
+- hosts: all
+  tasks:
+    - name: Include per-environment variables
+      include_vars: "vars/{{ app_env }}.yml"
+
+    - name: Set root prompt
+      lineinfile:
+        dest: /root/.bashrc
+        line: 'PS1="[{{prompt}}]$ "'
+        insertbefore: EOF
+YAML
+%>
+
+<%
+inventory_yaml=<<YAML
+[dev-servers]
+#{dev_server} app_env=#{dev_prompt}
+
+[prod-servers]
+prod1 app_env=dev
+prod2 app_env=dev
+YAML
+%>
+
+These variable files can be loaded via the [include_vars](https://docs.ansible.com/ansible/latest/modules/include_vars_module.html) module. Write the following playbook too <%=write_to_file('ansible/prompt.yml', prompt_playbook) %>
+```YAML
+<%= prompt_playbook %>
+```
+
+Here we can see that the `include_vars` module is being used to load the variables. The file that the variables should be loaded from are determined by inspecting the value of the `app_env` variable at run time. The `app_env` variable could be set against each server definition in the inventory for example. Write the following to <%= write_to_file('ansible/inventory', inventory_yaml) %>, which does just that.
+<% %>
+```YAML
+<%= inventory_yaml %>
+```
+
+Before running the playbook, run `cic connect <%= dev_server %>` and note that the prompt does contain the text `<%= dev_prompt %>`. After executing the playbook by running `<%=command("ansible-playbook ansible/prompt.yml -i ansible/inventory")%>`, connect back to <%= dev_server %> and you will see that the prompt now reads as required.
```

Comment:
Also... note that the prompt does not contain _dev_ .. but as the box is called dev1 then actually it does contain _dev_ ... `root@dev1:/cic#`

Reply:
Also, prompt after running exercise is uppercase `[DEV]$`

Reply:
I have fixed this so that DEV is written uppercase